src/nunavut/lang/c/templates/deserialization.j2

```diff
@@ -11,11 +11,16 @@
 
 {# ----------------------------------------------------------------------------------------------------------------- #}
 {% macro deserialize(t) %}
-    if ((out_obj == {{ valuetoken_null }}) || (buffer == {{ valuetoken_null }}) || {# -#}
-        (inout_buffer_size_bytes == {{ valuetoken_null }}))
+    if ((out_obj == {{ valuetoken_null }}) || (inout_buffer_size_bytes == {{ valuetoken_null }}) || {# -#}
+        ((buffer == {{ valuetoken_null }}) && (0 != *inout_buffer_size_bytes)))
     {
         return -NUNAVUT_ERROR_INVALID_ARGUMENT;
     }
+    else if (buffer == {{ valuetoken_null }})
+    {
+        *inout_buffer_size_bytes = 0;
+        return NUNAVUT_SUCCESS;
+    }
```

Comment:
This branch will cause invalid deserialization because it bypasses the implicit zero extension rule. Also, there should be an `else` after `else if`. Instead, I suggest this:

```suggestion
    if (buffer == {{ valuetoken_null }})
    {
        buffer = (const {{ typename_byte }}*)"";
    }
```